src/Data/AdditiveGroup.hs

```diff
@@ -31,16 +35,25 @@ import Foreign.C.Types (CSChar, CInt, CShort, CLong, CLLong, CIntMax, CFloat, CD
 
 import Data.MemoTrie
 
+import qualified GHC.Generics as Gnrx
+import GHC.Generics (Generic, (:*:)(..))
+
 infixl 6 ^+^, ^-^
 
 -- | Additive group @v@.
 class AdditiveGroup v where
   -- | The zero element: identity for '(^+^)'
   zeroV :: v
+  default zeroV :: (Generic v, AdditiveGroup (Gnrx.Rep v ())) => v
```

Comment:
Why `()`? Perhaps `Void` would fit better? And maybe use a type synonym, e.g., `type VRep v = Gnrx.Rep v Void`?